atd-etl/socrata_export/socrata_export.py

```diff
@@ -1,121 +1,95 @@
 
-# Define tables to query from Hasura and publish to Socrata
-query_configs = [
+datasets = [
     {
-        "table": "crash",
-        "template": crashes_query_template,
-        "formatter": format_crash_data,
-        "formatter_config": {
-            "tables": ["atd_txdot_crashes"],
-            "columns_to_rename": {
-                "veh_body_styl_desc": "unit_desc",
-                "veh_unit_desc_desc": "unit_mode",
-                "latitude_primary": "latitude",
-                "longitude_primary": "longitude",
-                "atd_fatality_count": "death_cnt",
-            },
-        },
-        "dataset_uid": ATD_ETL_CONFIG["SOCRATA_DATASET_CRASHES"],
+        "name": "crashes",
+        "typename": "socrata_export_crashes_view",
+        "dataset_id": SOCRATA_DATASET_CRASHES,
+        "query": QUERIES["crashes"],
     },
     {
-        "table": "person",
-        "template": people_query_template,
-        "formatter": format_person_data,
-        "formatter_config": {
-            "tables": ["atd_txdot_person", "atd_txdot_primaryperson"],
-            "columns_to_rename": {"primaryperson_id": "person_id"},
-            "prefixes": {
-                "person_id": "P",
-                "primaryperson_id": "PP",
-            },
-        },
-        "dataset_uid": ATD_ETL_CONFIG["SOCRATA_DATASET_PERSONS"],
+        "name": "people",
+        "typename": "socrata_export_people_view",
+        "dataset_id": SOCRATA_DATASET_PEOPLE,
+        "query": QUERIES["people"],
     },
 ]
 
-# Start timer
-start = time.time()
 
-# For each config, get records from Hasura and upsert to Socrata until res is []
-for config in query_configs:
-    print(f'Starting {config["table"]} table...')
-    is_truncate_complete = False
-    records = None
-    offset = 0
-    limit = 1000
-    total_records = 0
+def main(args_dict):
+    socrata_client = get_socrata_client()
 
-    # Query records from Hasura and upsert to Socrata
-    while records != []:
-        # Create query, increment offset, and query DB
-        query = config["template"].substitute(
-            limit=limit,
-            offset=offset,
-            date_limit=two_weeks_ago(),
-            initial_date_limit=ten_years_ago(),
-        )
-        offset += limit
-        data = run_hasura_query(query)
+    for dataset in datasets:
+        dataset_name = dataset["name"]
+        if not args_dict.get(dataset_name):
+            continue
 
-        if "data" not in data:
-            print("query: ", query)
-            print("data: ", data)
-            raise RuntimeError("There was a problem gathering data.")
+        logger.info(f"Processing {dataset_name}")
 
-        # Format records
-        records = config["formatter"](data, config["formatter_config"])
+        min_id = 0
+        records_processed = 0
+        while True:
+            logger.info(f"Fetching up to {RECORD_BATCH_SIZE} records")
+            variables = {"limit": RECORD_BATCH_SIZE, "minId": min_id}
```

Comment:
i initially implemented this using the `offset` flag, but it was extremely slow. querying by `id` with a record limit is much faster. 

interestingly, i also had to push the `order by id` clause down into the views. this was dramatically faster than querying the view with an `order` directive. it's a limitation of the pg query planner that i don't understand.